Sources/XcodeGraphMapper/Mappers/Packages/PackageMapper.swift

```diff
@@ -0,0 +1,196 @@
+import FileSystem
+import Foundation
+import Mockable
+import Path
+import XcodeGraph
+
+@Mockable
+protocol PackageMapping {
+    func map(
+        _ packageInfo: PackageInfo,
+        packages: [String: AbsolutePath],
+        at path: AbsolutePath
+    ) async throws -> Project
+}
+
+struct PackageMapper: PackageMapping {
+    private let fileSystem: FileSysteming
+
+    init(
+        fileSystem: FileSysteming = FileSystem()
+    ) {
+        self.fileSystem = fileSystem
+    }
+
+    func map(
+        _ packageInfo: PackageInfo,
+        packages: [String: AbsolutePath],
+        at path: AbsolutePath
+    ) async throws -> Project {
+        var targets: [Target] = []
+        for target in packageInfo.targets {
+            targets.append(
+                try await mapTarget(
+                    target,
+                    packageInfo: packageInfo,
+                    packages: packages,
+                    path: path
+                )
+            )
+        }
+        return Project(
+            path: path,
+            sourceRootPath: path,
+            xcodeProjPath: path,
+            name: packageInfo.name,
+            organizationName: nil,
+            classPrefix: nil,
+            defaultKnownRegions: nil,
+            developmentRegion: nil,
+            options: Project.Options(
+                automaticSchemesOptions: .disabled,
+                disableBundleAccessors: true,
+                disableShowEnvironmentVarsInScriptPhases: true,
+                disableSynthesizedResourceAccessors: true,
+                textSettings: Project.Options.TextSettings(
+                    usesTabs: nil,
+                    indentWidth: nil,
+                    tabWidth: nil,
+                    wrapsLines: nil
+                )
+            ),
+            settings: Settings(configurations: [:]),
+            filesGroup: .group(name: packageInfo.name),
+            targets: targets,
+            packages: [],
+            schemes: [],
+            ideTemplateMacros: nil,
+            additionalFiles: [],
+            resourceSynthesizers: [],
+            lastUpgradeCheck: nil,
+            type: .local
+        )
+    }
+
+    private func mapTarget(
+        _ target: PackageInfo.Target,
+        packageInfo: PackageInfo,
+        packages: [String: AbsolutePath],
+        path: AbsolutePath
+    ) async throws -> Target {
+        // Some of the products, such as "regular" are approximations until XcodeGraph supports these SwiftPM-specific products
```

Comment:
This is probably the weakest part of the PR. It might have been better to completely skip some targets like `.plugin`, but then they would not be represented for the hashing that should happen in selective testing. I'd rather not have ideal mapping here than skip those targets.